### PR TITLE
[1LP][RFR] generic object class level button tests

### PR DESCRIPTION
**Pull request by digitronik**
Signed-off-by: Nikhil Dhandre <ndhandre@redhat.com>

## Purpose or Intent
- Added edit related stuff
- adding custom button on generic object class level

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit bb0b34178f603786766b95488f07781a3d3c6b10

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_class_custom_button.py**
    - *test_custom_button_on_generic_class_crud*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by digitronik** [comment]
depends on #9228

**Comment by dajoRH** [comment]
I detected some fixture changes in commit d062c283545a45b800976b38dc0c865a69eea9ee

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_class_custom_button.py**
    - *test_custom_button_on_generic_class_crud*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 59d4fe54bbfb25b2b602a8e2d69a2edc3f4a06c9

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_class_custom_button.py**
    - *test_custom_button_on_generic_class_crud*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit a99da6f171043bf84b7d92fcc7b7aaffa194dee5

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_class_custom_button.py**
    - *test_custom_button_on_generic_class_crud*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit d45c894ab712f44e46115aea3cf6ff57ae9dc3a8

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_class_custom_button.py**
    - *test_custom_button_on_generic_class_crud*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/generic_objects/definition/button_groups.py [hunk 1]
```diff
@@ -174,14 +207,30 @@ def delete(self, cancel=False):
...
+        view.flash.assert_no_error()
+
+    def update(self, updates):
+        """Update this button in UI.
+
+        Args:
+            updates: Provided by update() context manager.
+        """
+        view = navigate_to(self, "Edit")
+        changed = view.fill(updates)
```
Comment (izapolsk) [inline cfme/generic_objects/definition/button_groups.py]: @digitronik, you reinvent the wheel here :).
please look at fill_with method()

Comment (digitronik) [reply cfme/generic_objects/definition/button_groups.py]: I was unknown about this method . its awesome... I need to changes on other places I will push PR in next sprint for this... @izapolsk++

**Comment by dajoRH** [comment]
I detected some fixture changes in commit cb96b7b52a7cef8ca10ebed065c540866428f6cc

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_class_custom_button.py**
    - *test_custom_button_on_generic_class_crud*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit c15919f9044714a91af98b6e96b074b3e095443e

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_generic_class_custom_button.py**
    - *test_custom_button_on_generic_class_crud*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by digitronik** [comment]
@izapolsk updated :+1:

**Pull request merged**